packages/plugins/src/plugins/spark/sparkYieldFetcher.ts

```diff
@@ -0,0 +1,79 @@
+import { EvmNetworkIdType, PortfolioElement } from '@sonarwatch/portfolio-core';
+import BigNumber from 'bignumber.js';
+import { getAddress } from 'viem';
+import { Cache } from '../../Cache';
+import {
+  CONVERSION_RATE_DIVISOR,
+  PLATFORM_ID,
+  ETHEREUM_NETWORK_ID,
+  SPARK_YIELD_POOLS_KEY,
+} from './constants';
+import { Fetcher, FetcherExecutor } from '../../Fetcher';
+import { ElementRegistry } from '../../utils/elementbuilder/ElementRegistry';
+import { SparkYieldPool } from './types';
+import { getBalances } from '../../utils/evm/getBalances';
+
+export default function getSparkYieldFetcher(
+  networkId: EvmNetworkIdType = ETHEREUM_NETWORK_ID
```

Comment:
We shouldn't provide a default value; the caller must provide it.

```suggestion
  networkId: EvmNetworkIdType
```